localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker_factory.py

```diff
@@ -111,6 +114,17 @@ def get_esm_worker(self) -> EsmWorker:
                 role_arn=self.function_role_arn,
                 service_principal=ServicePrincipal.lambda_,
                 source_arn=self.esm_config["FunctionArn"],
+                client_config=botocore.config.Config(
+                    retries={"total_max_attempts": 1},  # Disable retries
+                    read_timeout=max(
+                        self.esm_config.get(
+                            "MaximumBatchingWindowInSeconds", DEFAULT_MAX_WAIT_TIME_SECONDS
+                        ),
+                        60,
+                    )
+                    + 5,  # Extend read timeout (with 5s buffer) for long-polling
+                    tcp_keepalive=True,
```

Comment:
doc: some motivation and reference would be helpful on why we're doing this
Nice AWS reference: https://aws.amazon.com/blogs/networking-and-content-delivery/implementing-long-running-tcp-connections-within-vpc-networking/